cider/type/plan/UnaryExpr.cpp

```diff
@@ -24,6 +24,52 @@
 namespace Analyzer {
 using namespace cider::jitlib;
 
+JITValuePointer codegenCastBetweenDateAndTime(JITFunction& func,
+                                              JITValue& operand_val,
+                                              const SQLTypeInfo& operand_ti,
+                                              const SQLTypeInfo& ti) {
+  const int64_t operand_width = getTypeBytes(operand_ti.get_type());
+  const int64_t target_width = getTypeBytes(ti.get_type());
+  int64_t dim_scaled = DateTimeUtils::get_timestamp_precision_scale(
+      abs(operand_ti.get_dimension() - ti.get_dimension()));
+  JITValuePointer cast_scaled =
+      func.createConstant(JITTypeTag::INT64, dim_scaled * kSecondsInOneDay);
+  if (target_width == operand_width) {
+    return JITValuePointer(&operand_val);
+  } else if (target_width > operand_width) {
+    JITValuePointer cast_val =
+        operand_val.castJITValuePrimitiveType(getJITTypeTag(ti.get_type()));
+    return JITValuePointer(cast_val * cast_scaled);
```

Comment:
Do we need handle `NULL` property for cast?

Reply:
handle null value processing outside in `codegenCastFunc` func,  and have passed null value testcases in `CiderDateNextgenTest`